infrastructure/batch/roles.tf

```diff
@@ -63,7 +63,7 @@ resource "aws_iam_policy" "batch_ses_send_email" {
 EOF
 }
 
-resource "aws_iam_role_policy_attachment" "batch_ses_send_policy" {
+resource "aws_iam_role_policy_attachment" "ecs_task_ses_send_policy" {
```

Comment:
I don't think we need to attach the policy to the `ecs_task_role` at all. I would just move the policy down after the `batch_job_role` and keep the attachment there.